cg/meta/workflow/mutant/metrics_parser/metrics_parser.py

```diff
@@ -0,0 +1,76 @@
+from pathlib import Path
+from cg.io.csv import read_csv
+from typing import Any
+from cg.meta.workflow.mutant.metrics_parser.models import (
+    SampleResults,
+    SamplesResultsMetrics,
+)
+
+
+class MutantResultsHeaderData:
+    SAMPLE_NAME: str = "sample_name"
+    SELECTION: str = "selection"
+    REGION_CODE: str = "region_code"
+    TICKET: str = "ticket"
+    PCT_N_BASES: str = "pct_n_bases"
+    PCT_10X_COVERAGE: str = "pct_10x_coverage"
+    QC_PASS: str = "qc_pass"
+    LINEAGE: str = "lineage"
+    PANGOLIN_DATA_VERSION: str = "pangolin_data_version"
+    VOC: str = "voc"
+    MUTATIONS: str = "mutations"
```

Comment:
Humm could you please elaborate on how to set up the aliases? I'm not familiar with it. :/

Reply:
My take is that we essentially want a dictionary with keys being sample_internal_ids and each sample having a `SampleResults` value.
The `MetricsParser`, the `MutantResultsHeaderData` and the `MutantResultsHeaderRawData` all seem to have their part in this flow.

I claim that this can be done quicker and a bit more understandable by utilizing aliases. Consider writing:

```Python
class SampleResults(BaseModel):
    sample_name: str = Field(alias="Sample")
    selection: str = Field(alias="Selection")
    region_code: str = Field(alias="Region Code")
    ticket: Annotated[int | None, BeforeValidator(empty_str_to_none)] = Field(alias="Ticket")
    pct_n_bases: Annotated[float | None, BeforeValidator(empty_str_to_none)] = Field(alias="%N_bases")
    pct_10x_coverage: Annotated[float | None, BeforeValidator(empty_str_to_none)] = Field(alias="%10X_coverage")
    qc_pass: Annotated[bool, BeforeValidator(str_to_bool)] = Field(alias="QC_pass")
    lineage: str = Field(alias="Lineage")
    pangolin_data_version: str = Field(alias="Pangolin_data_version")
    voc: str = Field(alias="VOC")
    mutations: Annotated[str | None, BeforeValidator(empty_str_to_none)] = Field(alias="Mutations")
```

What this will do is create a Pydantic BaseModel where you have written that the parameter `sample_name` will be instantiated with 
a provided value called "Sample", that `selection` will be instantiated with the "Selection" value etc. You should be able to parse
the output from `read_csv` with the class above, although it can be shortened further by the use of a `TypeAdapter`. Basically,
if you have a list of Pydantic `BaseModels`, you can validate them all at the same time. The following code

```Python
raw_results: list[dict[Any, Any]] = [
                sample_results
                for sample_results in read_csv(file_path=results_file_path, read_to_dict=True)
            ]
adapter = TypeAdapter(list[SampleResults])
adapter.read_python(raw_results)
```
should at this point give you the list of `SampleResults` which you only need to convert to a dict with keys being the internal_ids.

Reply:
This sounds great!!! Thanks! Will implement!